### PR TITLE
Use tarfile python module for tar extraction

### PR DESCRIPTION
**Pull request by selaux**
Adresses #143.

I did not test the all builds so maybe that needs to be done before merging.

### CODE REVIEW

File: build.py [hunk 1]
```diff
@@ -43,12 +44,24 @@ def _rmtree_error_handler(func, path, exc_info):
     else:
         raise
 
+def _get_stripped_tar_members(tar):
```
Comment (pbor) [inline build.py]: the function could be local to Tarfile, no?

File: build.py [hunk 2]
```diff
@@ -43,12 +44,24 @@ def _rmtree_error_handler(func, path, exc_info):
     else:
         raise
 
+def _get_stripped_tar_members(tar):
+    for tarinfo in tar.getmembers():
+        path = tarinfo.name.split('/')
+        if len(path) == 1:
+            if tarinfo.isdir():
```
Comment (pbor) [inline build.py]: This could be shorter:

if len(path) == 1 and not tarinfo.isdir:
    throw

File: build.py (hunk 1 above)
Comment (selaux) [inline build.py]: Yes 🙂

File: build.py (hunk 2 above)
Comment (selaux) [inline build.py]: I dont think so, the continue is necessary to skip extraction of the top level directory.

**Comment by selaux** [comment]
Just noticed there might be an error with links, I am checking this currently.

**Comment by selaux** [comment]
Now tested with a lot of the included modules. Seems to work fine.

**Comment by guruDanny67** [comment]
Modified the build script to simply expand all the archive file in the build directory and tested with and without this modification and the two resulting build dirs are equals, so it seems the patch is ok :)

**Comment by nacho** [comment]
Thanks a lot!

**Comment by nacho** [comment]
Can you provide also a patch updating the README?

**Pull request merged**